framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -366,6 +368,35 @@ public AnnotatedTypeFactory(BaseTypeChecker checker) {
         }
     }
 
+    /**
+     * Issue an error and abort if any of the support qualifiers have @Target meta-annotations
+     * that contain something besides TYPE_USE or TYPE_PARAMETER. (@Target({}) is allowed)
+     */
+    private void checkSupportedQuals() {
+        for (Class<? extends Annotation> annotationClass : supportedQuals) {
+            ElementType[] elements = annotationClass.getAnnotation(Target.class).value();
+            String otherElementTypes = null;
```

Comment:
I would find the code easier to understand if otherElementTypes was a list, and then printing used String.join, rather than mixing the string processing code in with adding values to otherElementTypes.